File: products-api/internal/product/product.go
package product

type TaxCategory string

const (
	TaxCategoryTaxed   TaxCategory = "GRAVADO"
	TaxCategoryReduced TaxCategory = "REDUCIDO"
	TaxCategoryExempt  TaxCategory = "EXENTO"
)

type Product struct {
	ProductID     string      `json:"productId"`
	Name          string      `json:"name"`
	SKU           string      `json:"sku"`
	Category      string      `json:"category"`
	TaxCategory   TaxCategory `json:"taxCategory"`
	UnitOfMeasure string      `json:"unitOfMeasure"`
}

File: order-worker/src/main/java/com/b2b/orders/domain/model/Product.java
package com.b2b.orders.domain.model;

import java.util.Objects;

public record Product(
        String productId,
        String name,
        String sku,
        String category,
        TaxCategory taxCategory,
        String unitOfMeasure
) {
    public Product {
        requireText(productId, "productId");
        requireText(name, "name");
        requireText(sku, "sku");
        Objects.requireNonNull(taxCategory, "taxCategory is required");
    }

    private static void requireText(String value, String field) {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException(field + " is required");
        }
    }
}
